### PR TITLE
Feat(eos_designs): Adding port_channel_id as option for endpoint ethernet description

### LINKED ISSUE #4303: Support endpoint interfaces description to carry information which defined on other parts of the interfaces for Network ports in eos_designs role
### Enhancement summary

In our network, we are using the network ports format to configure the endpoints interfaces. Most of the ports are defined in port-channel pairly for 2x MLAG peers. Our endpoints will get the connected information from lldp (normally, descriptions/devices/interfaces) to decide which port-channel they will be in and according this information, they will get different configs on endpoints side. So we would like to include <channel_group_id> into the description.

By the way, for the automatically generated port-channel, we hope the port-channel id could be automatically included in description (By default each port will be configured in a port-channel with one member when leveraging automatic channel-id generation.) if we didn't define it. 

### Which component of AVD is impacted

eos_designs

### Use case example

```
  - switches:
      - switch-[ab].*
    switch_ports:
      - Ethernet1/1-2
    profile: xxx
    description: xxx_PoCH_<channel_group_id>
    port_channel:
      channel_id: 11
  ... ...
```
If we have the single port on each leaf pair member and didn't give the description key, the `channel_group_id` could be automatically added to the description:
```
  - switches:
      - switch-[ab].*
    switch_ports:
      - Ethernet1/1
    profile: xxx
```
Will be rendered as:
```
switch-a:
interface Ethernet1/1
  description Endpoints_Channel_Group_<channel_group_id>
  channel-group 11 mode active
  ... ...

switch-b:
interface Ethernet1/1
  description Endpoints_Channel_Group_<channel_group_id>
  channel-group 11 mode active
  ... ...
```

### Describe the solution you would like

Maybe define a var which can be used in description

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bjmeuer**
…ports ethernet intfs

## Change Summary

Adding endpoint_port_channel_id as option for endpoint ethernet description customisation. 

## Related Issue(s)

Fixes #4303 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
In the ethernet_interfaces.py in the get config function under the port-channel section adding the following:

```
# Update the description to make the port_channel_id usable
            ethernet_interface["description"] = (
                self.shared_utils.interface_descriptions.connected_endpoints_ethernet_interface(
                    InterfaceDescriptionData(
                        shared_utils=self.shared_utils,
                        interface=adapter["switch_ports"][node_index],
                        peer=peer,
                        peer_interface=peer_interface,
                        peer_type=connected_endpoint["type"],
                        description=interface_description,
                        port_channel_id=channel_group_id,
                    ),
                )
                or None
            )
```

With this the port-channel ID can be used in the ethernet_interfaces description.

## How to test

```
network_ports:
  - switches:
    - LEAF1
    switch_ports:
      - Ethernet2-3
    profile: MULTI
    mode: trunk
    vlans: 10,20
    description: "{endpoint_type!u}{endpoint?<_}{endpoint_port_channel_id?<_Po}"
```

```
  - profile: MULTI
    mode: trunk
    native_vlan: 99
    port_channel:
      mode: active
    ethernet_segment:
      short_esi: auto
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/api/interface_descriptions/__init__.py [hunk 1]
```diff
@@ -335,6 +335,7 @@ def connected_endpoints_ethernet_interface(self, data: InterfaceDescriptionData)
                     "endpoint": data.peer,
                     "endpoint_port": data.peer_interface,
                     "endpoint_type": data.peer_type,
+                    "endpoint_port_channel_id": data.port_channel_id,
```
Comment (gmuloc) [inline python-avd/pyavd/api/interface_descriptions/__init__.py]: Let's use the same variable name as the one in `connected_endpoints_port_channel_interface` for consistency

(line 401 of this file)

```suggestion
                    "port_channel_id": data.port_channel_id,
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml [hunk 2]
```diff
@@ -43,6 +43,7 @@ keys:
             The available template fields are:
               - `endpoint_type` - always set to `network_port`.
               - `endpoint` - content of the `endpoint` key if set.
+              - `endpoint_port_channel_id`: The value from `port_channel.channel_id` for this network port if set.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml]:
```suggestion
              - `port_channel_id`: The value from `port_channel.channel_id` for this network port if set.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_connected_endpoints.schema.yml [hunk 3]
```diff
@@ -64,3 +64,4 @@ $defs:
                     - `endpoint_type` - the `type` from `connected_endpoints_keys` like `server`, `router` etc.
                     - `endpoint` - The name of the connected endpoint
                     - `endpoint_port` - The value from `endpoint_ports` for this switch port if set.
+                    - `endpoint_port_channel_id`: The value from `port_channel.channel_id` for this switch port if set.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/defs_connected_endpoints.schema.yml]:
```suggestion
                    - `port_channel_id`: The value from `port_channel.channel_id` for this switch port if set.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/default_network_ports_description.yml [hunk 4]
```diff
@@ -16,6 +16,7 @@ keys:
       The available template fields are:
         - `endpoint_type`: Always set to `network_port`.
         - `endpoint`: The value of the `endpoint` key if set.
+        - `endpoint_port_channel_id`: The value from `port_channel.channel_id` for this network port if set.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/default_network_ports_description.yml]:
```suggestion
        - `port_channel_id`: The value from `port_channel.channel_id` for this network port if set.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/default_connected_endpoints_description.yml [hunk 5]
```diff
@@ -17,6 +17,7 @@ keys:
         - `endpoint_type`: The `type` from `connected_endpoints_keys` like `server`, `router` etc.
         - `endpoint`: The name of the connected endpoint
         - `endpoint_port`: The value from `endpoint_ports` for this switch port if set.
+        - `endpoint_port_channel_id`: The value from `port_channel.channel_id` for this switch port if set.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/default_connected_endpoints_description.yml]:
```suggestion
        - `port_channel_id`: The value from `port_channel.channel_id` for this switch port if set.
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml [hunk 6]
```diff
@@ -43,6 +43,7 @@ keys:
             The available template fields are:
               - `endpoint_type` - always set to `network_port`.
               - `endpoint` - content of the `endpoint` key if set.
+              - `port_channel_id`: The value from `port_channel.channel_id` for this network port if set.
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml]: Strictly speaking this is just the port-channel ID, and not necessarily coming from that variable. We auto-generate the ID if not set. Just copy the description from the port-channels.

File: python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/ethernet_interfaces.py [hunk 7]
```diff
@@ -176,6 +176,23 @@ def _get_ethernet_interface_cfg(self: AvdStructuredConfigConnectedEndpoints, ada
...
+                        interface=adapter["switch_ports"][node_index],
+                        peer=peer,
+                        peer_interface=peer_interface,
+                        peer_type=connected_endpoint["type"],
+                        description=interface_description,
+                        port_channel_id=channel_group_id,
+                    ),
+                )
+                or None
+            )
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/ethernet_interfaces.py]: Please incorperate it into the existion description generation instead. You can use `port_channel_id=channel_group_id if port_channel_mode is not None else None` and then move the assignment of port_channel_mode up.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml (hunk 6 above)
Comment (bjmeuer) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml]: done

**Comment by gmuloc** [comment]
@bjmeuer could you find a place in eos_designs_unit_test to use this new key and make sure it outputs what you want / need (you can override a description of some adapter / network_ports with port_channel)

**Review by ClausHolbechArista — APPROVED** [review]
LGTM apart from the missing test.

**Review by gmuloc — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM - reviewed test case

**Pull request merged**